src/Adapter/Product/ProductDataProvider.php

```diff
@@ -162,4 +163,32 @@ public function getImage($id_image)
             'base_image_url' => _THEME_PROD_DIR_ . $imageData->getImgPath(),
         ];
     }
+
+    /**
+     * Gets price with tax.
+     *
+     * @param int $productId
+     *
+     * @return float
+     */
+    public function getPriceWithTax(int $productId): float
+    {
+        return Product::getPriceStatic(
+            $productId,
+            true,
+            null,
+            Configuration::get('PS_PRICE_DISPLAY_PRECISION'),
```

Comment:
Please use Configuration Adapter.

Reply:
Hi, we already discussed this in https://github.com/PrestaShop/PrestaShop/pull/22731#discussion_r556368730

The product page is a big and complex topic 😉  so we plan to deliver it in multiple PRs, iterating over it.

@JevgenijVisockij tried to implement this suggestion but it made the scope of this PR much wider, which makes it more likely to break things and trigger a very long cycle of QA feedback. So it was restored to initial state.

Let's keep this part for a further PR. Small scoped PRs are what has worked the best for us so far when dealing with complex topics.